pubtools/_quay/quay_steps.py

```diff
@@ -0,0 +1,631 @@
+        "log_info": log info callback or None
+        "log_warning": log warning callback or None
+        "log_error": log error callback or None
+        "hub": pubhub xmlrpcserver instance
+        "task_id": task id
+        "target_name": target name
+    Produced details:
+        {
+            "items": { <repo>: { <tag> { "state": ready|error|pushed, "source": <pull_url>}}}
+    Results:
+        None
+    """
+
+    NAME = "StepPushOperators"
+
+    @staticmethod
+    def _get_immutable_tag(item):
+        """Return immutable tag from operator push item production tags."""
+        if item.metadata["v_r"] in list(item.metadata["tags"].values())[0]:
+            return item.metadata["v_r"]
+        # if v_r tag is not in destination tags, return tag with the most numbers in it
+        # This code is usually trigerred when push is not initiated from ET but manually
+        # by user
+        tags = []
+        for tag in list(item.metadata["tags"].values())[0]:
+            tags.append((len(re.split(r"\d+", tag)), tag))
+        return sorted(tags)[-1][1]
+
+    def _init_details(self):
+        push_items = self.external_resources["push_items"]
+        items = {}
+        self._details = {"items": items}
+        for item in push_items:
+            bundle_repo = list(item.metadata["destination"]["tags"].keys())[0]
+            bundle_ref = "{0}/{1}:{2}".format(
+                self.step_kwargs["docker_reference_registry"],
+                bundle_repo,
+                self._get_immutable_tag(item),
+            )
+            items[bundle_ref] = "ready"
+
+    def _pre_run(self):
+        if not self.step_kwargs.get("auto_upload_operators", True):
+            self.skip = True
+            self.skip_reason = "Automatic uploading of operators is not enabled"
+        if not self.step_kwargs.get("iib_server"):
+            self.skip = True
+            self.skip_reason = "IIB server is not defined"
+
+    def _push_operator_items(self, items, osev):
+        raise NotImplementedError
+
+    def pyxis_get_ocp_versions(self, item):
+        """
+        Get a list of supported ocp versions from Pyxis.
+
+        Args:
+            push_item: (_PushItem)
+                Push item for which the OCP version range will be found out.
+
+        Returns:
+            list(str): list of supported OCP versions as returned by Pyxis.
+        """
+        log_info = self.external_resources.get("log_info", lambda *args, **kwargs: ())
+        ocp_versions = item.metadata["com.redhat.openshift.versions"]
+        log_info("Getting OCP versions of '{0}' from Pyxis.".format(ocp_versions))
+        entry_point = (
+            "pubtools-pyxis",
+            "console_scripts",
+            "pubtools-pyxis-get-operator-indices",
+        )
+        entry_point_fn = pkg_resources.load_entry_point(*entry_point)
+        args = [
+            "cmd",
+            "--pyxis-server",
+            self.step_kwargs["pyxis_server"],
```

Comment:
This stuff is in target settings right? push_docker function doesn't produce this data

Reply:
fixed